### ISSUE TITLE
Upgrading applications with persistent storage

### ISSUE THREAD

**Issue by alahiff** [mesosphere/marathon#3597]
If I use the example json from here:

https://gist.github.com/gkleiman/598e4f778126e6af5b1c

to run PostgreSQL with a persistent volume, and then try to edit and deploy a change to the configuration (e.g. changing to a different version of the Docker image), I end up with the original task in the Waiting state and a new running task. However, there are two problems:

(1) the new running task is on a different host, attached to a new persistent volume, i.e. any data created by the first task is no longer being used

(2) the port mapping of the new task is incorrect: container port is always swapped with the service port (as reported in https://github.com/mesosphere/marathon/issues/3595)

I would have expected that with this upgrade strategy:

```
  "upgradeStrategy": {
    "maximumOverCapacity": 0,
    "minimumHealthCapacity": 0
  }
```

the original task would be killed, then a new task created on the same host, re-using the original persistent volume.

Is this a known issue or am I doing something wrong? I am using Mesos 0.28.0-2 and Marathon 1.0.0-RC1.

**Comment by gkleiman** [comment]
Looks related to #3603

**Comment by aquamatthias** [comment]
@meichstedt can you retest and make sure this works now?

**Comment by meichstedt** [comment]
Fixed in 22defe981f900027961edf3e42e538f2f3cdeb4c which will be part of Marathon v1.0.0-RC2

Background: Previously, The TaskReplaceActor always increased the maxCapacity by 1 `if (maxCapacity == app.instances)` which effectively set it to 2 in the case above with the described results. I cannot reproduce this behavior with docker image `mesosphere/marathon:latest-dev` given the above steps.
